src/core/structs.rs

```diff
@@ -365,16 +365,36 @@ impl_byte_array!(Uid);
 // NOTE TBZ: need struct to implement `Serializable`
 #[must_use]
 #[derive(Default, Debug, Clone)]
-pub struct EncryptedTable<const UID_LENGTH: usize>(HashMap<Uid<UID_LENGTH>, Vec<u8>>);
+pub struct EncryptedTable<const UID_LENGTH: usize>(Vec<(Uid<UID_LENGTH>, Vec<u8>)>);
```

Comment:
we could declare an `EncryptedTableLine` struct for the type `(Uid<UID_LENGTH>, Vec<u8>)`